app/views/spree/orders/_line_item.html.haml

```diff
@@ -25,8 +26,13 @@
         = line_item.unit_price_price_and_unit
     %td.text-center.cart-item-quantity{"data-hook" => "cart_item_quantity"}
       - finalized_quantity = @order.completed? ? line_item.quantity : 0
+      - max_quantity = variant.on_demand && 9999 || variant.on_hand
```

Comment:
~~I realise this was existing logic, but it contains a hardcoded number and makes a decision about stock availabilty. So I think it should be moved out of the view and into the model or a stock-related Concern. 
The good news is that we already have one - `VariantStock`. It already has a similar method `can_supply?`.~~

But.. this might all be unecessary if my below comment about Stimulus Reflex is valid.

Reply:
I did not refactor this part, I've only factorize one variable that was already filled in the `ofn-on-hand` attribute.
As far as I understand `VariantStock::can_supply?` returns a boolean that indicates if we can supply the variant, although the `variant.on_demand && 9999 || variant.on_hand` returns the max value the user can buy. Have I missed something?

Reply:
You're absolutely right, I was just pointing out that this `max_quantity` logic could be a new method added to `VariantStock`, because it makes a decision about how to count stock. 
However I think I was being too picky. This is fine as it is.